### ISSUE TITLE
Replace default scalar values across all argument types

### ISSUE THREAD

**Issue by VaishakVk** [anvilco/spectaql#147]
Hi, 

I am trying to replace the default static values that are part of `spectaql`. I tried adding `metadata.json` with `SCALAR` and also tried using `dynamicExamplesProcessingModule`. Below is the config file for `scalarProcessor`. 

```
function scalarProcessor(argz = {}) {
  const { name } = argz;
  switch (name) {
    case 'Int': {
      // Life, the Universe and Everything
      return 42;
    }
    case 'String': {
      // Life, the Universe and Everything
      return 'example';
    }
    case 'BigInt': {
      // Life, the Universe and Everything
      return 42;
    }
    case 'Timestamp': {
      // Life, the Universe and Everything
    }
    case 'Float': {
      // Life, the Universe and Everything
      return 42;
    }
  }
}
```

However, it just replaces the static type and all the arguments are not replaced with the value provided. Is there a way to update all the scalar values present inside arguments as well. 

eg - `{"id": 987.65, "name": "abc123"}`

I tried logging `argumentProcessor` and `fieldProcessor` to see if it was possible to update defaults in argument.

**Comment by newhouse** [comment]
Hello @VaishakVk and thanks for using SpectaQL! Sorry this got lost for so long. Just to be sure I understand you correctly:
1. You'd like to specify the **_example_** value used for various Scalars (as it seems like you have figured out already).
2. You'd like to use those same example values wherever any of those Scalars are used as Arguments.
3. I presume you'd also like to use those same example values wherever any of those Scalars are used as Fields in your Types.


You should be able to use something like this to get what you want:
```js
const _ = require('lodash')

function handleScalar (name) {
  switch (name) {
    case 'Int': {
      return 42;
    }
    case 'String': {
      return 'fourty-two';
    }
    case 'BigInt': {
      return 42;
    }
    case 'Timestamp': {
      return "\"1978-11-27T01:08:03.420Z\""
    }
    case 'Float': {
      return 42.00;
    }
    default: {
      return
    }
  }
}

function scalarProcessor (argz = {}) {
  const {
    definition,
    name,
  } = argz
  // Don't stomp on an example if there already is one
  if (typeof _.get(definition, 'example') !== 'undefined') {
    return
  }

  return handleScalar(name)
}

function fieldProcessor (argz = {}) {
  const {
    returnType,
    definition,
  } = argz

  // Don't stomp on an example if there already is one
  if (typeof _.get(definition, 'properties.return.example') !== 'undefined') {
    return
  }

  return handleScalar(returnType)
}

function argumentProcessor (argz = {}) {
  const {
    definition,
    type,
  } = argz

  // Don't stomp on an example if there already is one
  if (typeof _.get(definition, 'example') !== 'undefined') {
    return
  }

  return handleScalar(type)
}

module.exports = {
  fieldProcessor,
  argumentProcessor,
  scalarProcessor,
}
```

Can you try this out and let me know how that works for you?

**Comment by VaishakVk** [comment]
Thank you for answering. I tried your solution, however for `Types` the defaults don't seem to substitute. Attached an image that was generated after adding the example module. 
![image](https://user-images.githubusercontent.com/30084393/143921086-cf357703-4c8e-42ea-9027-4ca64459b888.png)

The `id` column has a `Float` type but the example still shows the default value even after the example module was setup as the above message in the thread.

**Comment by newhouse** [comment]
@VaishakVk seems like you're actually experiencing this on a Field on an Input Type? Is that the only place where things are not working as expected?

Can you provide me with a simplified SDL/Schema that I can use to reproduce this? And can you call out what parts exactly are not working as expected, even with the code I provided?

**Comment by VaishakVk** [comment]
Yes @newhouse , It happens only on Input fields.  Ive created a repo to help debugging - https://github.com/VaishakVk/spectaql-debug 

You could run `yarn` and `yarn doc` to generate the documentation.

**Comment by newhouse** [comment]
Ok @VaishakVk I've just merged [this PR](https://github.com/anvilco/spectaql/pull/166) and published the changes in `v0.10.0` that add an [`inputFieldProcessor` hook](https://github.com/anvilco/spectaql/blob/main/examples/customizations/examples/index.js#L102) to the set of examples processors that you already know about.

If you update / pull the latest version of the code, you should be able to add an `inputFieldProcessor` handler that follows the pattern of the other handlers, which will give you what you're looking for.

Can you try that out, please?

**Comment by VaishakVk** [comment]
Thank you, works good now. Closing the issue.
